core/engine/src/object/builtins/lazy_builtin.rs

```diff
@@ -0,0 +1,451 @@
+use super::JsFunction;
+use crate::{
+    gc::custom_trace,
+    object::{
+        internal_methods::{
+            non_existant_call, non_existant_construct, ordinary_define_own_property,
+            ordinary_delete, ordinary_get, ordinary_get_own_property, ordinary_get_prototype_of,
+            ordinary_has_property, ordinary_is_extensible, ordinary_own_property_keys,
+            ordinary_prevent_extensions, ordinary_set, ordinary_set_prototype_of, ordinary_try_get,
+            CallValue, InternalMethodContext, InternalObjectMethods,
+        },
+        JsPrototype,
+    },
+    property::{PropertyDescriptor, PropertyKey},
+    realm::{Realm, RealmInner},
+    Context, JsData, JsNativeError, JsObject, JsResult, JsValue,
+};
+use boa_gc::{Finalize, Trace, WeakGc};
+use boa_string::JsString;
+use std::cell::Cell;
+
+#[derive(Debug, Clone, Trace, Finalize)]
+pub(crate) enum BuiltinKind {
+    Function(JsFunction),
```

Comment:
Hmm, thinking a bit more about this, I think `Function` should store a `NativeFunctionObject` instead, and the builder methods should be adjusted to consider this. Else, the builder won't set the builtin to the correct value while initializing.